azure_sdk/patches/03_missing_includes.patch

```diff
@@ -0,0 +1,12 @@
+diff -ur '--exclude=*.git' azure-iot-sdk-c-lts_01_2022/c-utility/adapters/socketio_berkeley.c azure-iot-sdk-c-lts_01_2022-phoenix/c-utility/adapters/socketio_berkeley.c
+--- azure-iot-sdk-c-lts_01_2022/c-utility/adapters/socketio_berkeley.c	2022-06-29 14:12:04.364923641 +0200
++++ azure-iot-sdk-c-lts_01_2022-phoenix/c-utility/adapters/socketio_berkeley.c	2022-06-29 14:17:03.679322382 +0200
+@@ -7,6 +7,8 @@
+ #endif
+ 
+ #define _DEFAULT_SOURCE
++/* struct sockaddr is provided by socket.h, there were an assumption that if.h includes it */
++#include <sys/socket.h>
```

Comment:
Posix requires `sockaddr` structure to be present in `sys/socket.h`
https://pubs.opengroup.org/onlinepubs/9699919799.2018edition/

Reply:
Ok, now I remeber where the problem is. We can't include `sys/socket.h` in `net/if.h`, since lwip uses its own sockets structures definitions. It would be nice to fix this, since for example newlib does include `sys/socket.h` in `net/if.h`, but we can live with it. Even GNU says, that on some platforms `net/if.h` is not self-contained:
https://www.gnu.org/software/gnulib/manual/html_node/net_002fif_002eh.html

Reply:
Nitpicking: `there was an assumption`.